node-gyp fails on mac osx el capitan with only mono

`which` doesn't return empty string when a command isn't found:
![jskrzypek_jss-macbook-pro____dev_tjanczuk_edge](https://cloud.githubusercontent.com/assets/1513265/12874641/007aa9fa-cde0-11e5-96aa-980cbc8f104b.png)

This means that node-gyp will think `dnx` is available, even when its not and you get this output/error:

```sh
gyp info it worked if it ends with ok
gyp info using node-gyp@3.2.1
gyp info using node@4.2.6 | darwin | x64
gyp info spawn /usr/local/bin/python2
gyp info spawn args [ '/Users/jskrzypek/.nvm/versions/node/v4.2.6/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/jskrzypek/dev/tjanczuk/edge/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/jskrzypek/.nvm/versions/node/v4.2.6/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/jskrzypek/.node-gyp/4.2.6/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/jskrzypek/.node-gyp/4.2.6',
gyp info spawn args   '-Dnode_gyp_dir=/Users/jskrzypek/.nvm/versions/node/v4.2.6/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/jskrzypek/dev/tjanczuk/edge',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/clractioncontext.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/clrfunc.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/clrfuncinvokecontext.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/monoembedding.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/task.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/dictionary.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/nodejsfunc.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/nodejsfuncinvokecontext.o
  CXX(target) Release/obj.target/edge_nativeclr/src/mono/utils.o
  CXX(target) Release/obj.target/edge_nativeclr/src/common/edge.o
  SOLINK_MODULE(target) Release/edge_nativeclr.node
  CXX(target) Release/obj.target/edge_coreclr/src/common/v8synchronizationcontext.o
  CXX(target) Release/obj.target/edge_coreclr/src/common/edge.o
  CXX(target) Release/obj.target/edge_coreclr/src/CoreCLREmbedding/coreclrembedding.o
  CXX(target) Release/obj.target/edge_coreclr/src/CoreCLREmbedding/coreclrfunc.o
  CXX(target) Release/obj.target/edge_coreclr/src/CoreCLREmbedding/coreclrnodejsfunc.o
  CXX(target) Release/obj.target/edge_coreclr/src/CoreCLREmbedding/coreclrfuncinvokecontext.o
  CXX(target) Release/obj.target/edge_coreclr/src/CoreCLREmbedding/coreclrnodejsfuncinvokecontext.o
  CXX(target) Release/obj.target/edge_coreclr/src/common/utils.o
  SOLINK_MODULE(target) Release/edge_coreclr.node
  ACTION binding_gyp_build_managed_target_restore_packages src/CoreCLREmbedding/project.lock.json
bash: dnu: command not found
make: *** [src/CoreCLREmbedding/project.lock.json] Error 127
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/jskrzypek/.nvm/versions/node/v4.2.6/lib/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.3.0
gyp ERR! command "/Users/jskrzypek/.nvm/versions/node/v4.2.6/bin/node" "/Users/jskrzypek/.nvm/versions/node/v4.2.6/bin/node-gyp" "configure" "build"
gyp ERR! cwd /Users/jskrzypek/dev/tjanczuk/edge
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok
```

--- a/binding.gyp
+++ b/binding.gyp
@@ -58,7 +58,7 @@
           {
             'conditions': [
               [
-                '"<!(echo -n `which dnx`)"!=""',
+                '"<!((which dnx) || echo not_found)"!="not_found"',
                 {
                   'sources+': [
                     'src/common/v8synchronizationcontext.cpp',
@@ -162,7 +162,7 @@
           {
             'conditions': [
               [
-                '"<!(echo -n `which mono`)"!=""',
+                '"<!((which mono) || echo not_found)"!="not_found"',
                 {
                   'sources+': [
                     'src/mono/clractioncontext.cpp',
@@ -286,7 +286,7 @@
           {
             'conditions': [
               [
-                '"<!(echo -n `which mono`)"!=""',
+                '"<!((which mono) || echo not_found)"!="not_found"',
                 {
                   'actions+': [
                     {
@@ -310,7 +310,7 @@
                 }
               ],
               [
-                '"<!(echo -n `which dnx`)"!=""',
+                '"<!((which dnx) || echo not_found)"!="not_found"',
                 {
                   'actions+': [
                     {

